src/alchemy/alchemy.service.ts

```diff
@@ -156,42 +156,39 @@ export class AlchemyService {
         }
     }
 
-    async getTokenHistoricPrices(tokenAddress: string, startTimes: string[],interval: string): Promise<{ price: number; date: string }[]> {
-        const url = `https://api.g.alchemy.com/prices/v1/${this.apiKey}/tokens/historical`;
-        const results: { price: number; date: string }[] = [];
-        
-        for (const startTime of startTimes) {
-            const beginDate = new Date(startTime);
-            const formattedBeginDate = beginDate.toISOString();
-            const formattedEndDate = new Date().toISOString();
-
-            const data = {
-                startTime: formattedBeginDate,
-                endTime: formattedEndDate,
-                interval: interval,
-                address: tokenAddress,
-                network: "base",
-            };
-
-            try {
-                const response = await axios.post(url, data);
-                const priceArray = response.data.data.map((item: any) => ({
-                    price: item.value,
-                    date: item.timestamp,
-                }));
-                results.push(...priceArray);
-            } catch (error) {
-                if (error.response) {
-                    console.error(`API returned an error:`, error.response.data);
-                } else if (error.request) {
-                    console.error(`No response received from API:`, error.request);
-                } else {
-                    console.error(`Error setting up request:`, error.message);
+    async getTokenHistoricPrices(symbol: string = "WETH", startTimes: string="2024-11-21T00:00:00Z", interval: string="1d"): Promise<any> {
+        const options = {
+            method: 'POST',
+            headers: {accept: 'application/json', 'content-type': 'application/json'},
+            body: JSON.stringify({
+              symbol: symbol,
+              startTime: startTimes,
+              endTime: new Date().toISOString(),
+              interval: interval
+            })
+          };
```

Comment:
![image](https://github.com/user-attachments/assets/dcf1bc36-057a-4889-9ba6-a9819bc179c5)
a discussion i had with ben about the use of symbol or token address, tbf the function is super clean but i'd align with ben's knowledge on this ...

Reply:
I think it's not an issue knowing that this function will only be call by us and never automatically